server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregator.java

```diff
@@ -155,23 +112,23 @@ public InternalGeoGrid buildAggregation(long owningBucketOrdinal) throws IOExcep
         consumeBucketsAndMaybeBreak(size);
 
         BucketPriorityQueue ordered = new BucketPriorityQueue(size);
```

Comment:
I think, now that `ordered` only ever carries one bucket type, we can update it to be:

```
BucketPriorityQueue<InternalGeoGridBucket> ordered = new BucketPriorityQueue<>(size);
```

that way we do not need to cast `(InternalGeoGridBucket)` in later places, right?

Reply:
:+1: 

As an aside, it looks like we can remove the generics on GeoGridAggregator if we wanted to.  E.g. just use `InternalGeoGridBucket` everywhere, instead of `<B extends InternalGeoGridBucket>`. 

Not going to make that change in this PR, and there might be places where it is still useful, but wanted to mention :)